js/components/matrix.js

```diff
@@ -0,0 +1,363 @@
+    getOptionId: function(rowIndex, columnIndex, type) {
+      return _.kebabCase(this.$parent.id + '-' + this.name + '-' + rowIndex + '-' + columnIndex + '-' + type);
+    },
+
+    /**
+     * Returns unique name to the form widget instance, field, row and column.
+     *
+     * @param {Number} rowIndex - an row index
+     *
+     * @returns {String} an name unique to the form widget instance, field, row and column
+     */
+    getOptionName: function(rowIndex) {
+      return _.kebabCase(this.name + '-' + rowIndex);
+    },
+
+    /**
+     * Click handler for each radio button in matrix
+     *
+     * @param {Object} row - an row object which includes id and label
+     *
+     * @param {Object} column - an column object which includes id and label
+     *
+     * @param {Number} rowIndex - an row index
+     *
+     * @param {Number} colIndex - an column index
+     *
+     * @returns {undefined}
+     */
+    clickHandler: function(row, column, rowIndex, colIndex) {
+      if (rowIndex >= 0 && colIndex >= 0) {
+        var el = this.getOptionId(rowIndex, colIndex, 'input');
+
+        $('#' + el).prop('checked', true);
+      }
+
+      if (row.id && this.value[row.id]) {
+        delete this.value[row.id];
+      }
+
+      if (this.value[row.label]) {
+        delete this.value[row.label];
+      }
+
+      this.value[row.id || row.label] = column ? column.id || column.label : undefined;
+
+      this.$emit('_input', this.name, this.value);
+    },
+
+    /**
+     * Focus handler for each radio button in matrix
+     *
+     * @param {Number} rowIndex - an row index
+     *
+     * @param {Number} colIndex - an column index
+     *
+     * @returns {undefined}
+     */
+    focusHandler: function(rowIndex, colIndex) {
+      var $vm = this;
+
+      if (rowIndex >= 0 && colIndex >= 0) {
+        var row = this.rowOptions[rowIndex];
+        var col = this.columnOptions[colIndex];
+
+        $('#' + $vm.getOptionId(rowIndex, colIndex, 'input')).focus();
+
+        $('#' + $vm.getOptionId(rowIndex, colIndex, 'span')).focus();
+
+        this.value[row.id || row.label] = col.id || col.label;
+        this.clickHandler(row, col, rowIndex, colIndex);
+      }
+    },
+
+    /**
+     * Set default undefined value for each row if value is blank
+     *
+     * @returns {undefined}
+     */
+    setDefaultValue: function() {
+      var $vm = this;
+
+      $vm.value = {};
+
+      _.forEach(this.rowOptions, function(row) {
+        $vm.$set($vm.value, row.id || row.label, undefined);
+      });
+    },
+
+    /**
+     * Set value for each row as per selection and default value
+     *
+     * @returns {undefined}
+     */
+    setValue: function() {
+      var $vm = this;
+
+      if ($vm.value === undefined || _.isEmpty($vm.value)) {
+        this.setDefaultValue();
+      } else {
+        if (typeof $vm.value === 'string') {
+          $vm.value = JSON.parse($vm.value);
+        }
+
+        _.forOwn($vm.value, function(key, value) {
```

Comment:
We'd usually use `_.forIn()` in this case. What's the reason for choosing to use `_.forOwn()`?

```suggestion
        _.forIn($vm.value, function(key, value) {
```

Reply:
Hi @tonytlwu  _.forOwn() Method Iterates over the only own keys of the given object and invoke iteratee for each property. Where as  _.forIn() Method Iterates over both keys and values of the given object and invokes iteratee function for each property, that is why I used _.forOwn, but as you suggested I will replace with _.forIn.